<?xml version="1.0"?>
<launch>
   <arg name="viz" default="off"/>
   <arg name="filter" default="off"/>

   <include file="$(find uav_sim_ros)/launch/manual_teleop.launch"/>
   <include file="$(find uav_sim_ros)/launch/rviz.launch" if="$(eval viz == 'on')"/>

	<group ns="uav_sim_ros">

		<node name="integrator" pkg="uav_sim_ros" type="integrator_node" output="screen">
			<param name="dt" value="0.001" />
			<rosparam param="IC">[0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]</rosparam>
		</node>

		<node name="safety_filter" pkg="uav_sim_ros" type="safety_filter_node" output="screen">
			<param name="pass_through" value="0" if="$(eval filter == 'on')"/>
			<param name="pass_through" value="1" if="$(eval filter == 'off')"/>
			<param name="integration_dt" value="0.01"/>
			<param name="backup_Tmax" value="1"/>
			<param name="safety_buffer" value="0.2"/>
			<param name="safety_buffer_soft" value="0.2"/>
			<param name="terminal_vel_max" value="0.05"/>
			<param name="tau_backup" value="0.1"/>
			<param name="smoothing_tau_obstacle" value="0.2"/>
			<param name="smoothing_tau_vDes" value="0.1"/>

			<param name="hoverThrust" value="0.5127" />
			<param name="KiVz" value="5.0" />
			<param name="KpVz" value="1.0" />
			<param name="KpVxy" value="0.3" />
			<param name="KpAttitude" value="0.5" />
			<param name="KdAttitude" value="0.05" />
			<param name="KpOmegaz" value="2.0" />
			<param name="maxInclination" value="30.0" />
		</node>

		<node name="controller" pkg="uav_sim_ros" type="controller_node" output="screen">
			<param name="hoverThrust" value="0.52" />
			<param name="KiVz" value="5.0" />
			<param name="KpVz" value="1.0" />
			<param name="KpVxy" value="0.1" />
			<param name="KpAttitude" value="0.5" />
			<param name="KdAttitude" value="0.05" />
			<param name="KiOmegaz" value="1.0" />
			<param name="KpOmegaz" value="2.0" />
			<param name="maxInclination" value="30.0" />
		</node>

	</group>
</launch>
